### PR TITLE
Add feature flag package

### PR DESCRIPTION
**Pull request by andyg0808**
This is an attempt at an ultra-simple feature flag package which could be used across the monorepo and this NextJS repo.

See the readme for rationale and design notes.

The goal is to add a minimal package which we can use to consistently have feature flags available.

qa_req 0
This code is unused.

Connects https://github.com/iFixit/ifixit/issues/50604

### CODE REVIEW

File: packages/feature_flags/index.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+   if (flagValue === true) {
+      return true;
+   }
+
+   const enableFlag = 'enable-' + flagName;
+
+   if (typeof flagValue === 'object') {
+      if (flagValue.userToggle) {
+         return localStorage.getItem(enableFlag) === 'true';
```
Comment (aburke07) [inline packages/feature_flags/index.ts:17]: Do we need to check if localstorage is available? Also, should this be `window.localstorage`?

```suggestion
         return window.localStorage.getItem(enableFlag) === 'true';
```

File: packages/feature_flags/index.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+   return false;
+}
+
+function getHashBoolean(param: string): boolean {
+   const hash = window?.location?.hash;
+   if (!hash) {
+      return false;
+   }
+   return hash.indexOf(param) !== -1;
+}
```
Comment (aburke07) [inline packages/feature_flags/index.ts:35]: Do we also want to support query params to enable flags?

**Review by aburke07 — APPROVED** [review]
CR :alien:

Dev_block :alien: on two comments

File: packages/feature_flags/index.ts (hunk 2 above)
Comment (andyg0808) [reply packages/feature_flags/index.ts:35]: No, possibly not-yet. I don't have a strong opinion one way or the other, and I want this minimal. Using the hash guarantees that nothing different gets sent to the server.

**Comment by aburke07** [comment]
Un_dev_block :alien:

**Pull request merged**